lib/streamlit/elements/radio.py

```diff
@@ -77,7 +89,7 @@ def radio(
         self,
         label: str,
         options: OptionSequence[T],
-        index: int = 0,
+        index: Optional[Union[int, None]] = 0,
```

Comment:
`Optional[Union[int, None]]` is a bit redundant since `Optional[int]` is the same as `Union[int, None]`. I would suggest to directly use the newer annotation form: `int | None` here instead. But you need to first import this: `from __future__ import annotations` on the top of the file.

Reply:
Thank you @LukasMasuch , I've followed your recommendations for all the widgets and replaced old notation with a new one ex. `int | None` 👍 the PR is updated